src/dsp_tools/analyse_xml_data/construct_and_analyze_graph.py

```diff
@@ -299,13 +301,13 @@ def generate_upload_order(
         cycle = list(rx.digraph_find_cycle(graph))  # type: ignore[attr-defined]  # pylint: disable=no-member
         links_to_remove = _find_cheapest_outgoing_links(graph, cycle, edges)
         stash_counter += len(links_to_remove)
-        links_to_stash = _remove_edges_to_stash(
+        _remove_edges_to_stash(
             graph=graph,
             edges_to_remove=links_to_remove,
             all_edges=edges,
             remaining_nodes=remaining_node_indices,
         )
-        stash_lookup = _add_stash_to_lookup_dict(stash_lookup, links_to_stash)
+        stash_lookup = _add_stash_to_lookup_dict(stash_lookup, [x.link_object for x in links_to_remove])
```

Comment:
I preferred the previous version: both more readable and most likely faster

Reply:
The reason I took it out of `_remove_edges_to_stash()` is: This function removes the edges, as its name says. But then, it also extracted the link objects from a parameter it received: `[x.link_object for x in links_to_remove]` was done inside the function, and the result was returned. 

That's completely unnecessary, this can be done on the caller side. The old version was misleading: The function did not only what its name said, but it also extracted something from one of its parameters, and returned it.

I don't think that this would be faster or more readable. But perhaps it's a misunderstanding?

Reply:
Let me clarify:
- I think it's more readable to have a named list than a comprehension
- speed: internally, this list that previously has been returned is still created inside the function, previously that was just returned, now you re-create it on the caller side. My assumption - without testing it - is that creating this list twice is slower than passing it back from the function.

Reply:
- I agree that a named list can be more readable than a comprehension.
- But I disagree that the list is still created inside the function: The list `[x.link_object for x in links_to_remove]` is completely unrelated to what happens in `_remove_edges_to_stash()`. Inside there, the `ResptrLink | XMLLink`s are completely ignored. So I found it misleading to create a list with `ResptrLink | XMLLink`s inside that function, and then return it. 

Or am I completely confusing things?

Reply:
maybe I'm confused... but from what I see:

previously:

```python
links_to_stash = [x.link_object for x in edges_to_remove]
...
for link_to_stash in links_to_stash:
    ...
...
return links_to_stash
```

now:

```python
...
for link_to_stash in [x.link_object for x in edges_to_remove]:
    ....
...
```
so I think you just in-lined the list creation into the for loop, discard the list because you don't hold on to it; and then on the caller side, you create the same list again.

Reply:
Oh I see, you are right. So from performance side, my code will be slower, because the same list is built 2 times.

I didn't realize it, because if you analyze it from a "flow of information" view, then it doesn't make sense that a function called `_remove_edges_to_stash()` returns an information that can easily be extracted from a parameter that you pass to that function.

So, I still think that my version is "nicer", but the previous is faster. I'll text you in chat concerning performance measurements.

Reply:
sounds good, thanks!